modules/basic/ds/hashmap.h

```diff
@@ -217,6 +225,262 @@ class HashmapBuilder : public HashmapBaseBuilder<K, V, H, E> {
+
+  /**
+   * @brief Reserve the size for the hashmap.
+   *
+   */
+  void reserve(size_t size) {
+    LOG(INFO) << __func__ << " is not finished.";
+    vec_kv_.reserve(size);
+  }
+
+  /**
+   * @brief Return the maximum possible size of the HashMap, i.e., the number
+   * of elements that can be stored in the HashMap.
+   *
+   */
+  size_t bucket_count() const {
+    // return hashmap_.bucket_count();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return 0;
+  }
+
+  /**
+   * @brief Return the load factor of the HashMap.
+   *
+   */
+  float load_factor() const {
+    // return hashmap_.load_factor();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return 0;
+  }
+
+  /**
+   * @brief Check whether the hashmap is empty.
+   *
+   */
+  bool empty() const {
+    // return hashmap_.empty();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return true;
+  }
+
+  /**
+   * @brief Return the beginning iterator.
+   *
+   */
+  void* begin() {
+    // return hashmap_.begin();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the const beginning iterator.
+   *
+   */
+  void* begin() const {
+    // return hashmap_.begin();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the const beginning iterator.
+   *
+   */
+  void* cbegin() const {
+    // return hashmap_.cbegin();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the ending iterator
+   *
+   */
+  void* end() {
+    // return hashmap_.end();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the const ending iterator.
+   *
+   */
+  void* end() const {
+    // return hashmap_.end();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the const ending iterator.
+   *
+   */
+  void* cend() const {
+    // return hashmap_.cend();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Find the value by key.
+   *
+   */
+  void* find(const K& key) {
+    // return hashmap_.find(key);
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Associated with a given data buffer
+   */
+  void AssociateDataBuffer(std::shared_ptr<Blob> data_buffer) {
+    // this->data_buffer_ = data_buffer;
+  }
+
+  template <typename K_ = K>
+  typename std::enable_if<std::is_integral<K_>::value, void>::type Construct(
+      int concurrency = 1) {
+    size_t count = 0;
+    vec_kv_.resize(n_elements_);
+    vec_k_.resize(vec_kv_.size());
+    uint64_t start_time = GetCurrentTime();
+    for (auto& kv_ : vec_kv_) {
+      vec_k_[count] = kv_.first;
+      count++;
+    }
+    LOG(INFO) << "Constructing the vec_k_ takes "
+              << GetCurrentTime() - start_time << " s.";
```

Comment:
Use VLOG(100) for profiling related logs. See my revision to your varint encoding PR.

Reply:
Done